src/EFCore.PG/Infrastructure/Internal/NpgsqlOptionsExtension.cs

```diff
@@ -286,9 +287,29 @@ public virtual NpgsqlOptionsExtension WithRemoteCertificateValidationCallback([C
         }
 
         #endregion Authentication
+
+        public override long GetServiceProviderHashCode()
+        {
+            unchecked
+            {
+                if (_serviceProviderHash == null)
+                {
+                    _serviceProviderHash = _userRangeDefinitions.Aggregate(
+                        base.GetServiceProviderHashCode(),
+                        (h, ud) => (h * 397) ^ ud.GetHashCode());
+                    _serviceProviderHash = (_serviceProviderHash * 397) ^ AdminDatabase?.GetHashCode() ?? 0L;
+                    _serviceProviderHash = (_serviceProviderHash * 397) ^ (PostgresVersion?.GetHashCode() ?? 0L);
+                    _serviceProviderHash = (_serviceProviderHash * 397) ^ (ProvideClientCertificatesCallback?.GetHashCode() ?? 0L);
+                    _serviceProviderHash = (_serviceProviderHash * 397) ^ (RemoteCertificateValidationCallback?.GetHashCode() ?? 0L);
+                    _serviceProviderHash = (_serviceProviderHash * 397) ^ ReverseNullOrdering.GetHashCode();
```

Comment:
There is the wonderful `HashCode` type. Why not use it? (:

Reply:
I... somehow missed this class entirely! Am definitely going to switch to it.

Reply:
Oh, I see why I haven't heard of it - it seems to have been added in .NET Core 2.1... There seems to be [an OOB nuget](https://www.nuget.org/packages/Microsoft.Bcl.HashCode) but it doesn't really seem worth it... Will leave the code like this unless you feel strongly about it.

Reply:
And I just remembered that since EF Core switched to .NET Standard 2.1, so will we... Let's merge this as-is for now, and we can do a later pass to convert all hash code generation to use this type, OK?

Reply:
I'm fine with it.